store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -8189,6 +8221,53 @@ public void modifyContact(OperationContext octxt, int contactId, ParsedContact p
         }
         return newAddrs;
     }
+    
+    /**
+     * Returns a list of contacts don't exist in any emailed contacts folders.
+     * @param octxt OperationContext object
+     * @param addrs Recipient Email address list
+     * @return listOfContactsToBeModified This list consist list of Map, email as key and Pair of ContactId and 
+     * DisplayName as Value
+     */
+    public List<Map<String, Pair<Integer, String>>> findContactsInEmailedContacts
+    (OperationContext octxt, Collection<Address> addrs) {
+        if (addrs.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<Map<String, Pair<Integer, String>>> list = new ArrayList<>();
+        Map<String, Pair<Integer, String>> map = new HashMap<>();
+
+        for (Address addr : addrs) {
+            if (addr instanceof javax.mail.internet.InternetAddress) {
+                javax.mail.internet.InternetAddress iaddr = (javax.mail.internet.InternetAddress) addr;
+                try {
+                    if (!Strings.isNullOrEmpty(iaddr.getAddress())) {
+                        ZimbraQueryResults results = index.search(octxt, "in:\"Emailed Contacts\" field[email]:" + iaddr.getAddress(),
```

Comment:
should we use id instead of folder name? ID_FOLDER_AUTO_CONTACTS

Reply:
Done, thanks!

Reply:
Changes have been made in modifyAutoContact method when language is set to Japanese.